autopilot/migrator.go

```diff
@@ -198,29 +198,20 @@ func (m *migrator) performMigrations(p *workerPool) {
 	default:
 	}
 
-OUTER:
-	for {
-		// fetch currently configured set
-		autopilot, err := m.ap.Config(m.ap.shutdownCtx)
-		if err != nil {
-			m.logger.Errorf("failed to fetch autopilot config: %w", err)
-			return
-		}
-		set := autopilot.Config.Contracts.Set
-		if set == "" {
-			m.logger.Error("could not perform migrations, no contract set configured")
-			return
-		}
-
-		// recompute health.
-		start := time.Now()
-		if err := b.RefreshHealth(m.ap.shutdownCtx); err != nil {
-			m.ap.RegisterAlert(m.ap.shutdownCtx, newRefreshHealthFailedAlert(err))
-			m.logger.Errorf("failed to recompute cached health before migration: %v", err)
-			return
-		}
-		m.logger.Infof("recomputed slab health in %v", time.Since(start))
+	// fetch currently configured set
```

Comment:
In practice this probably doesn't work. Not that that's an issue we have to fix here but most likely if you have a huge backlog of migrations, updating the autopilot config to change set would never have any effect migration-wise because we'd be "stuck" migrating the old backlog and the maintenance finished signals would come so fast we'd just keep updating the set with slabs from the previously configured set.

Reply:
When changing the set, the endpoint interrupts the maintenance. So this should be interrupted and then restart with the new set.